x-pack/plugins/cloud_security_posture/public/common/api/use_compliance_dashboard_data_api.ts

```diff
@@ -7,14 +7,36 @@
 
 import { type QueryObserverOptions, useQuery } from '@tanstack/react-query';
 import { useKibana } from '../hooks/use_kibana';
-import { ComplianceDashboardData } from '../../../common/types';
+import { PosturePolicyTemplate, ComplianceDashboardData } from '../../../common/types';
 import { STATS_ROUTE_PATH } from '../../../common/constants';
 
-const getStatsKey = ['csp_dashboard_stats'];
+const getCspmStatsKey = ['csp_cspm_dashboard_stats'];
+const getKspmStatsKey = ['csp_kspm_dashboard_stats'];
 
-export const useComplianceDashboardDataApi = (
+export const getStatsRoute = (policyTemplate: PosturePolicyTemplate) => {
+  return STATS_ROUTE_PATH.replace('{policy_template}', policyTemplate);
+};
+
+export const useCspmComplianceDashboardDataApi = (
+  options: QueryObserverOptions<unknown, unknown, ComplianceDashboardData, unknown, string[]>
```

Comment:
i don't think this is the correct type. `QueryObserver` is more lower-level than `UseQueryOptions`

it's usually pretty hard to make generic `useQuery` hooks due to many generics. i would avoid trying tbh and only pass what's needed.